### PR TITLE
[CODEMOD:FN app-directory-boilerplate] RootLayout not created when project does not have _document file

### PR DESCRIPTION
**Pull request by DmytroHryshyn**
closes https://linear.app/intuita/issue/INT-2076/[codemodfn-app-directory-boilerplate]-rootlayout-not-created-when

### CODE REVIEW

File: next/13/app-directory-boilerplate/index.ts [hunk 1]
```diff
@@ -50,6 +50,20 @@ export default function NotFound() {
...
+export default function RootLayout({
+	children,
+  }: {
+	children: React.ReactNode
+  }) {
+	return (
+	  <html lang="en">
+		<body>{children}</body>
+	  </html>
+	)
```
Comment (hbjORbj) [inline next/13/app-directory-boilerplate/index.ts:63]:
```suggestion
	);
```

File: next/13/app-directory-boilerplate/index.ts [hunk 2]
```diff
@@ -50,6 +50,20 @@ export default function NotFound() {
...
+  }
```
Comment (hbjORbj) [inline next/13/app-directory-boilerplate/index.ts:64]:
```suggestion
  };
```

File: next/13/app-directory-boilerplate/index.ts [hunk 3]
```diff
@@ -1016,12 +1035,17 @@ const handleData: HandleData<Dependencies, State> = async (
...
 		}
 
-		if (
-			filePurpose === FilePurpose.ROOT_LAYOUT &&
-			options.underscoreDocumentData
-		) {
+		if (filePurpose === FilePurpose.ROOT_LAYOUT) {
 			const { tsmorph } = api.getDependencies();
 
+			if (options.underscoreDocumentData === undefined) {
```
Comment (hbjORbj) [inline next/13/app-directory-boilerplate/index.ts:1041]: `underscoreDocumentData` is string or null, no?

Comment (DmytroHryshyn) [reply next/13/app-directory-boilerplate/index.ts:1041]: its string or undefined

File: next/13/app-directory-boilerplate/index.ts [hunk 4]
```diff
@@ -869,24 +879,33 @@ const handleFile: Filemod<
...
-					name: 'layout',
-				}),
-				options: {
-					...options,
-					underscoreDocumentPath,
-					underscoreDocumentData,
-					filePurpose: FilePurpose.ROOT_LAYOUT,
-				},
-			});
+		let underscoreDocumentData: string | null = null;
```
Comment (hbjORbj) [inline next/13/app-directory-boilerplate/index.ts:884]: then this should be modified accordingly

**Review by hbjORbj — APPROVED** [review]

**Pull request merged**